tests/test_optimizer/unittest_fused_adam.py

```diff
@@ -0,0 +1,68 @@
+import torch
+import torch.nn as nn
+from torch.optim.adam import Adam
+from torch.optim import AdamW
+
+from colossalai.nn.optimizer.fused_adam import FusedAdam
+from colossalai.testing import parameterize
+
+
+class FC(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+        self.fc = nn.Sequential(nn.Linear(1024, 1024))
+    def forward(self, x):
+        return self.fc(x)
+
+
+@parameterize('adamw', [True, False])
+@parameterize('p_dtype', [torch.float, torch.half])
+@parameterize('g_dtype', [torch.float, torch.half])
+def test_adam(adamw, p_dtype, g_dtype):
+    model = FC().cuda().to(p_dtype)
+    state = model.state_dict()
+    model_copy = FC().cuda()
+    model_copy.load_state_dict(state.copy())
+    model_copy.to(torch.float)
+
+    if adamw:
+        optim = FusedAdam(model.parameters(), weight_decay=1e-2, adam_w_mode=True)
+        torch_optim = AdamW(model_copy.parameters(), weight_decay=1e-2)
+    else:
+        optim = FusedAdam(model.parameters(), weight_decay=1e-2)
+        torch_optim = Adam(model_copy.parameters(), weight_decay=1e-2)
+
+    data = torch.rand(10, 1024).cuda().to(p_dtype)
+    data_copy = data.clone().float()
+    label = torch.rand(10, 1024).cuda().to(p_dtype)
+    label_copy = label.clone().float()
+
+    for i in range(len(data)):
+        y = model(data[i])
+        loss = ((label[i] - y) ** 2).sum()
+        optim.zero_grad()
+        loss.backward()
+        if p_dtype != g_dtype:
+            for i in range(len(optim.param_groups[0]['params'])):
+                optim.param_groups[0]['params'][i].grad.data = optim.param_groups[0]['params'][i].grad.data.to(g_dtype)
+        optim.step()
+
+    for i in range(len(data_copy)):
+        y = model_copy(data_copy[i])
+        loss = ((label_copy[i] - y) ** 2).sum()
+        torch_optim.zero_grad()
+        loss.backward()
+        torch_optim.step()
+
+    assert len(optim.param_groups[0]['params']) == len(torch_optim.param_groups[0]['params'])
+    max_diff = 0
+    for i in range(len(optim.param_groups[0]['params'])):
+        diff = torch.max(torch.abs(optim.param_groups[0]['params'][i]-(torch_optim.param_groups[0]['params'][i])))
```

Comment:
Is it possible to use `torch.allclose()`?

Reply:
yes, thanks for your advice.